c/planarityApp/planarityCommandLine.c

```diff
@@ -6,8 +6,11 @@ See the LICENSE.TXT file for licensing information.
 
 #include "planarity.h"
 
-#if defined(_MSC_VER) && !defined(__INTEL_COMPILER)
+#if defined(_MSC_VER) && !defined(__llvm__) && !defined(__INTEL_COMPILER)

```

Comment:
The change to add "!defined(__llvm__)" triggers two questions. 

The first is just why was it necessary to add it? In other  words, do we ever have _MSC_VER defined and __INTEL COMPILER undefined for any compiler other than the Windows Microsoft cl compiler? 

The second is a question I should've asked earlier, which is why must __INTEL COMPILER be undefined? In other words, is _MSC_VER ever defined for any compiler other than the Windows Microsoft cl compiler? In other other words, is _MSC_VER being defined sufficient to uniquely detect the Microsoft cl compiler, or does testing only for _MSC_VER being defined lead to the `#if` part for any other compiler (gcc on any platform, with or without VSCode, or clang with VSCode)?

Reply:
According to the [`cpredef`](https://github.com/cpredef/predef/tree/master) [`Compilers.md`](https://github.com/cpredef/predef/blob/master/Compilers.md) documentation, we need to make sure `!defined(__INTEL_COMPILER)` because the Intel compiler defines `_MSC_VER`. Similarly, if `clang` is run on Windows as `clang.exe`, then `_MSC_VER` is defined, so we need `!defined(__llvm__) && !defined(__INTEL_COMPILER)` to make sure that block is only being run when we're building using MSVC `cl`. There might be a more streamlined test, but I figured this was the most comprehensive!!